theano/gof/opt.py

```diff
@@ -1167,39 +1174,39 @@ def transform(self, node):
             missing.difference_update(givens.keys())
         # ensure we have data for all input variables that need it
         if missing:
-            if self.verbose:
+            if self.verbose > 0:
                 print(("%s cannot meta-optimize %s, "
                        "%d of %d input shapes unknown" %
                        (self.__class__.__name__, node, len(missing), node.nin)))
             return
         # now we can apply the different optimizations in turn,
         # compile the resulting subgraphs and time their execution
-        if self.verbose:
+        if self.verbose > 1:
             print(("%s meta-optimizing %s (%d choices):" %
-                   (self.__class__.__name__, node, len(self.optimizers))))
+                   (self.__class__.__name__, node, len(self.get_opts(node)))))
         timings = []
-        for opt in self.optimizers:
+        for opt in self.get_opts(node):
             outputs = opt.transform(node)
             if outputs:
                 try:
                     fn = theano.function([], outputs, givens=givens,
                                          on_unused_input='ignore')
                     timing = min(self.time_call(fn) for _ in range(3))
```

Comment:
In the new back-end, and with cudnn conv "time_*" mode, the first call will be much slower. As there is 3 call, I'm not sure it is worth to make another one before, but in reality, there will probably be only the 2 second that will give sensible timing.

Also, to help distinguish more the difference, you could some of Theano function overhead by doing:
fn.trust_input = True
before using it.